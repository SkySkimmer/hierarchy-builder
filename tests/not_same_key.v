From HB Require Import structures.

HB.mixin Record A T := { a : T }.
#[key="T"]
HB.mixin Record B T1 (T : Type) := { b : T -> T1 }.

Fail HB.structure Definition sAB T1 := {T of A T & B T T1}.