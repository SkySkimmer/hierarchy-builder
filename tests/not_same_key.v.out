The command has indeed failed with message:
HB: all mixins must have the same key
